src/Sulu/Bundle/SearchBundle/Controller/SearchController.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SearchBundle\Controller;
+
+use Massive\Bundle\SearchBundle\Search\SearchManagerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use FOS\RestBundle\View\ViewHandler;
+use FOS\RestBundle\View\View;
+use JMS\Serializer\SerializationContext;
+use Pagerfanta\Adapter\ArrayAdapter;
+use Pagerfanta\Pagerfanta;
+
+/**
+ * Sulu search controller
+ */
+class SearchController
+{
+    /**
+     * @var SearchManagerInterface
+     */
+    private $searchManager;
+
+    /**
+     * @var ViewHandler
+     */
+    private $viewHandler;
+
+    /**
+     * @param SearchManagerInterface $searchManager
+     */
+    public function __construct(
+        SearchManagerInterface $searchManager,
+        ViewHandler $viewHandler
+    )
+    {
+        $this->searchManager = $searchManager;
+        $this->viewHandler = $viewHandler;
+    }
+
+    /**
+     * Perform a search and return a JSON response
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $query = $request->query->get('q');
+        $category = $request->query->get('category', null);
+        $locale = $request->query->get('locale', null);
+        $page = $request->query->get('page', 1);
+        $pageSize = $request->query->get('page_size', 50);
+        $query = $this->searchManager->createSearch($query);
```

Comment:
For this kind of stuff there would be the `ListRestHelper`, would also make sure that the same names for these parameters are used.